identity-service/src/relay/txRelay.js

```diff
@@ -242,6 +216,51 @@ const sendTransactionInternal = async (req, web3, txProps, reqBodySHA) => {
   return txReceipt
 }
 
+const filterReplicaSetUpdates = (decodedABI, senderAddress) => {
+  // Rate limit replica set reconfiguration transactions
+  // A reconfiguration (as opposed to a first time selection) will have an
+  // _oldPrimaryId value of "0"
+
+  let isReplicaSetTransaction = false
+  let isFirstReplicaSetConfig = false
+
+  if (decodedABI.name === 'updateReplicaSet') {
+    // TODO remove legacy replica set updates
+    isReplicaSetTransaction = decodedABI.name === 'updateReplicaSet'
+    if (isReplicaSetTransaction) {
+      isFirstReplicaSetConfig = decodedABI.params.find(
+        (param) => param.name === '_oldPrimaryId' && param.value === '0'
+      )
+    }
+  } else if (decodedABI.name === 'EntityManager') {
+    isReplicaSetTransaction = decodedABI.params.find(
+      (param) =>
+        param.name === '_entityType' && param.value === 'UserReplicaSet'
+    )
+    // isFirstReplicaSetConfig must be false
+    // EntityManager create user actions include the initial replica set
+  }
+
+  if (!isFirstReplicaSetConfig) {
+    transactionRateLimiter.updateReplicaSetReconfiguration += 1
+    if (
+      transactionRateLimiter.updateReplicaSetReconfiguration >
+      UPDATE_REPLICA_SET_RECONFIGURATION_LIMIT
+    ) {
+      throw new Error('updateReplicaSet rate limit reached')
+    }
+
+    if (
```

Comment:
should the whitelist check happen before the rate limit one? otherwise we are incrementing the rate limit incorrectly

Reply:
yea i think you're right

Reply:
actually i think this is intentional from https://github.com/AudiusProject/audius-protocol/pull/4457

> Adds an optional whitelist for updateReplicaSet for state machine updateReplicaSet ops. This only applies after all existing rules run like is update replica set check, rate limit

Reply:
you're right actually, should keep this way